Kitodo-DataFormat/src/main/java/org/kitodo/dataformat/access/MultiMapCollector.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.dataformat.access;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+/**
+ * Collects an object stream to a multi-map. A multi-map is a
+ * {@code Map<K, Set<V>>}, which can store several objects under one key.
+ */
+class MultiMapCollector<T, K, V> implements Collector<T, Map<K, Set<V>>, Map<K, Set<V>>> {
```

Comment:
Why don't you use an own class for this?
this:
```
Collector<T, Map<K, Set<V>>, Map<K, Set<V>>>
```
seems to be a bit confusing.

Reply:
The interface is so defined in Java, I have no influence on that.